src/analysis/processing/qgsalgorithmextractlabels.cpp

```diff
@@ -0,0 +1,372 @@
+
+      if ( dataDefinedValues.contains( QgsPalLayerSettings::MultiLineHeight ) )
+      {
+        QgsTextFormat format = labelSettings.format();
+        format.setLineHeight( dataDefinedValues.value( QgsPalLayerSettings::MultiLineHeight ).toDouble() );
+        labelSettings.setFormat( format );
+      }
+
+      QgsTextFormat format = labelSettings.format();
+      double formatLineHeight = format.lineHeight();
+
+      QgsAttributes attributes;
+      attributes << mMapLayerNames->value( layerId ) << fid << labelText << labelAlignment << labelRotation
+                 << fontFamily << fontStyle << fontSize << fontItalic << fontBold << fontLetterSpacing << fontWordSpacing
+                 << formatLineHeight;
+
+      double x = label->getX();
+      double y = label->getY();
+      if ( dataDefinedValues.contains( QgsPalLayerSettings::PositionX ) )
+      {
+        x = dataDefinedValues.value( QgsPalLayerSettings::PositionX ).toDouble();
+      }
+      if ( dataDefinedValues.contains( QgsPalLayerSettings::PositionY ) )
+      {
+        y = dataDefinedValues.value( QgsPalLayerSettings::PositionY ).toDouble();
+      }
+
+      QgsGeometry geometry( new QgsPoint( x, y ) );
+
+      QgsFeature feature;
+      feature.setAttributes( attributes );
+      feature.setGeometry( geometry );
+      features << feature;
+    };
+
+    QMap<QString, QString> *mMapLayerNames = nullptr;
+    QgsProcessingFeedback *mFeedback = nullptr;
+    QList<QgsFeature> features;
+};
+
+QVariantMap QgsExtractLabelsAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  // MAP_THEME has been taken care of in prepareAlgorithm
+  QgsCoordinateReferenceSystem crs = parameterAsExtentCrs( parameters, QStringLiteral( "EXTENT" ), context );
+  if ( !crs.isValid() )
+    crs = context.project()->crs();
+  const QgsRectangle extent = parameterAsExtent( parameters, QStringLiteral( "EXTENT" ), context );
+  const double scale = parameterAsDouble( parameters, QStringLiteral( "SCALE" ), context );
+  const double dpi = parameterAsDouble( parameters, QStringLiteral( "DPI" ), context );
+
+  QgsScaleCalculator calculator;
+  calculator.setDpi( dpi );
+  calculator.setMapUnits( crs.mapUnits() );
+  const QSizeF imageSize = calculator.calculateImageSize( extent, scale );
+
+  QgsFields fields;
+  fields.append( QgsField( QStringLiteral( "layer" ), QVariant::String, QString(), 0, 0 ) );
+  fields.append( QgsField( QStringLiteral( "featureid" ), QVariant::LongLong, QString(), 20 ) );
+  fields.append( QgsField( QStringLiteral( "label_text" ), QVariant::String, QString(), 0, 0 ) );
+  fields.append( QgsField( QStringLiteral( "label_alignment" ), QVariant::String, QString(), 0, 0 ) );
+  fields.append( QgsField( QStringLiteral( "label_rotation" ), QVariant::Double, QString(), 20, 10 ) );
+  fields.append( QgsField( QStringLiteral( "font_family" ), QVariant::String, QString(), 0, 0 ) );
```

Comment:
Can we use the equivalent name from the matching data defined property here for the field names? This would then make it possible to easily create fields from ALL labeling settings in future without breaking models and other workflows.